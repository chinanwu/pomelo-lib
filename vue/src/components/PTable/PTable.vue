<template>
  <table>
    <thead>
      <tr>
        <th v-for="(header, index) in headers" :key="'PTableHd-' + index">
          {{ format(header) }}
        </th>
      </tr>
    </thead>
    <tbody>
      <tr v-for="(item, index) in data" :key="'PTableTr-' + index">
        <td v-for="(value, key) in item" :key="'PTableTd-' + index + key">
          {{ value }}
        </td>
      </tr>
    </tbody>
  </table>
</template>

<script>
import capitalize from "@/functions/capitalize.js";

export default {
  name: "PTable",
  props: {
    headers: {
      type: Array,
      required: true,
      validator: values => values.every(value => typeof value === "string")
    },
    data: {
      type: Array,
      default: null,
      validator: values => values.every(value => typeof value === "object")
    }
  },
  methods: {
    format(str) {
      return capitalize(str);
    }
  }
};
</script>

<style lang="less" scoped>
table,
th,
td {
  border: 1px solid black;
}
</style>
