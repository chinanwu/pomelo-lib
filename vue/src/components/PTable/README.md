# PTable

A Table component

[Click here to see a demo of these buttons in action](https://chinanwu.github.io/pomelo-lib/vue/index.html#ptable)

## Example
```html
<PTable
    headers="['headerone', 'headertwo']"
    data="[{ headerone: 'something', headertwo: 2}]"
></PTable>
```
Alternatively, you can use the hyphenated component name.
```html
<p-table></p-table>
```

## Props

| Prop | Type |  Default | Required | Description | Validation |
| ---- | ---- | -------- | -------- | ----------- | ---------- |
| headers | Array | N/A | true | An array of table headers | Must be an array of string |
| data | Array | null | false | An array of table contents | Must be an array of objects |

## Best practices

- The number of headers and the number of keys in each object in data should be the same. i.e. if you have 6 headers, each object should have 6 keys.

## See an issue?
- Please create an issue [here](https://github.com/chinanwu/pomelo-lib/issues). Indicate the component and the issue, thank you! 
- Alternatively, if you fix the issues, you can create a [pull request](https://github.com/chinanwu/pomelo-lib/pulls)! 
