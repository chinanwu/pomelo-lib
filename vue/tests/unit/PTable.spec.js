import { shallowMount } from "@vue/test-utils";

import PTable from "@/components/PTable/PTable";

describe("PTable", () => {
  describe("rendering", () => {
    it("renders PTable", () => {
      const wrapper = shallowMount(PTable);
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
