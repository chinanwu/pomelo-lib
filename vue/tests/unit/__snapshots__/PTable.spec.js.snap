// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PTable rendering renders PTable 1`] = `
<table>
  <thead>
    <tr>
      
      
    </tr>
  </thead>
  <tbody>
    
    
  </tbody>
</table>
`;
